skipthought/model.py

```diff
@@ -77,12 +79,12 @@ def _create_decoder(self, scope_name, encoder_state, decoder_input):
             b = tf.get_variable("proj_b", [self.max_vocab_size])
             output_projection = (w, b)
 
-            decoder_outputs, _ = tf.nn.seq2seq.rnn_decoder(embedded_prev, initial_state=encoder_state,
-                                                           cell=cell)
+            decoder_outputs, _ = tf.contrib.legacy_seq2seq.rnn_decoder(embedded_prev, initial_state=encoder_state, cell=cell)
 
-        loop_function_predict = tf.nn.seq2seq._extract_argmax_and_embed(self.embedding_matrix, output_projection=(w, b), update_embedding=False)
+        from tensorflow.contrib.legacy_seq2seq.python.ops.seq2seq import _extract_argmax_and_embed
+        loop_function_predict = _extract_argmax_and_embed(self.embedding_matrix, output_projection=(w, b), update_embedding=False)
         with tf.variable_scope(scope_name, reuse=True):
-            decoder_predict_hiddens, _ = tf.nn.seq2seq.rnn_decoder(embedded_prev, initial_state=encoder_state,
+            decoder_predict_hiddens, _ = tf.contrib.legacy_seq2seq.rnn_decoder(embedded_prev, initial_state=encoder_state,
```

Comment:
I think it's better not to use legacy code. Look at  `tf.contrib.seq2seq.dynamic_rnn_decoder`. It has parameter `decoder_fn` which is actually `loop_function` in legacy version. Could you modify the code in that way?